stdlib/private/SwiftPrivateLibcExtras/Subprocess.swift

```diff
@@ -13,14 +13,19 @@
 import SwiftPrivate
 #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
 import Darwin
-#elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku)
+#elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku) || os(WASI)
 import Glibc
 #elseif os(Windows)
 import MSVCRT
 import WinSDK
 #endif
 
 internal func _signalToString(_ signal: Int) -> String {
+  #if os(WASI)
+  // no signals support on WASI yet, see https://github.com/WebAssembly/WASI/issues/166
+  fatalError("\(#function) is not supported on WebAssembly/WASI")
```

Comment:
I don't think that you should stub out `_signalToString`.  Instead, you need to die earlier at the point where the signal is being handled.  Just drop the entire function on WASI.